include/nuttx/board.h

```diff
@@ -429,6 +429,23 @@ int board_composite_initialize(int port);
 FAR void *board_composite_connect(int port, int configid);
 #endif
 
+/****************************************************************************
+ * Name:  board_usbdev_serialstr
+ *
+ * Description:
+ *   Use board unique serial number string to iSerialNumber field in the
+ *   device descriptor. This is for determining the board when multiple
+ *   boards on the same host.
+ *
+ * Returned Value:
+ *   The board unique serial number string.
+ *
+ ****************************************************************************/
+
+#if defined(CONFIG_BOARD_USBDEV_SERIALSTR)
+FAR const char *board_usbdev_serialstr(void);
```

Comment:
Yes, but I don't recommend to use board_uniqueid directly. Because USB serial number can use any string, `MYDEV112233` for example. This new API is for possible to do that.

Reply:
But should we give a more general name here? It's too bad to couple serial string to usbdev. I could prefer that:
#if defined(CONFIG_BOARD_UNIQUESTR)
FAR const char *board_uniquestr(void);
#endif
and provide a defined implementation on top of board_uniqueid.